src/dataAccessLayer.ts

```diff
@@ -0,0 +1,64 @@
+import "reflect-metadata";
+
+import { Block, State, Transaction } from "@blockr/blockr-models";
+import { DataSource } from "./clients";
+import DIContainer from "./injection/container";
+import { LevelBlockchainRepository, LevelStateRepository} from "./repositories";
+import { MongoBlockchainRepository, MongoStateRepository } from "./repositories";
+import { IBlockchainRepository, IStateRepository } from "./repositories";
+
+export class DataAccessLayer {
+    private blockchainRepository: IBlockchainRepository;
+    private stateRepository: IStateRepository;
+
+    constructor(dataSource: DataSource) {
+        if (dataSource === DataSource.LEVEL_DB) {
```

Comment:
Perhaps a `switch` statement would be more beneficial here? Even though there are only two data sources available as of now, using a `switch` statement turns this code into future-proof code; when we'd add a new data source, all we'd have to do is add a new `case` to the switch, rather than change the entire logic of this constructor. The `if-statement` causes the constructor to only support two data sources, rather than infinite.